cmd/test-utils_test.go

```diff
@@ -83,6 +83,9 @@ func init() {
 // concurreny level for certain parallel tests.
 const testConcurrencyLevel = 10
 
+// mutex for concurrent access to global variables
+var globalVarMu sync.RWMutex
```

Comment:
> This locking is not needed @vadmeste we shouldn't be modifying global variables in that manner, the approach taken in tests is fine.

```
WARNING: DATA RACE
Write at 0x000002909e18 by goroutine 276:
  github.com/minio/minio/cmd.testAdminCmdRunnerSignalService()
      /home/travis/gopath/src/github.com/minio/minio/cmd/admin-rpc_test.go:44 +0x94

 Previous read at 0x000002909e18 by goroutine 194:
  github.com/minio/minio/cmd.testServiceSignalReceiver()
      /home/travis/gopath/src/github.com/minio/minio/cmd/admin-handlers_test.go:467 +0x70
```

Most likely this happens because cmd.testServiceSignalReceiver() is run in a goroutine, similar to what some people understood in stackoverflow.

Basically tests don't run in parallel, but if one test runs a goroutine without shutting down it properly, it can remain working and it will be conflicted with another test.

Reply:
This I understand but its a haphazard way of doing locking like this, having some sort of a common locker for all global variables. There is no well-defined approach on how it should be accessed etc.

In generall i see that this test is useless, it is introducing a data race.

Reply:
Yes, I am reverting the code and send the fix in a new PR which should fix the data race